### PR TITLE
fix(projenrc): projenrc class does not create `--outdir`

### LINKED ISSUE #2099: Projenrc class does not create `--outdir`
This is perhaps intentional, but I can't find any documentation either way, and since I can pass `--outdir` when using `projen new`, I assume this is bug, but maybe not?

I want to initialize a new projen project, but in a subdirectory, and the `Projenrc` class expects the path to exist, which in my opinion, if you're using `projen new` to init a new project, is an unexpected requirement.

```
zmully@ubu2 /tmp/tmp.PquOlmUf69 $ cd $(mktemp -d) && npx projen new cdktf-construct --outdir cdktf
node:internal/fs/utils:345
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/tmp/tmp.WKMGJYxSdl/cdktf/.projenrc.js'
    at Object.openSync (node:fs:585:3)
    at Object.writeFileSync (node:fs:2170:35)
    at Projenrc.generateProjenrc (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/javascript/projenrc.js:46:14)
    at new Projenrc (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/javascript/projenrc.js:19:14)
    at new NodeProject (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/javascript/node-project.js:216:13)
    at new TypeScriptProject (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/typescript/typescript.js:20:9)
    at new JsiiProject (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/cdk/jsii-project.js:46:9)
    at new ConstructLibrary (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/cdk/construct-lib.js:12:9)
    at new ConstructLibraryCdktf (/home/zmully/.npm/_npx/80cfb0dc84733c29/node_modules/projen/lib/cdktf/cdktf-construct.js:19:9)
    at evalmachine.<anonymous>:1:27 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/tmp/tmp.WKMGJYxSdl/cdktf/.projenrc.js'
}
zmully@ubu2 /tmp/tmp.WKMGJYxSdl $ mkdir cdktf
zmully@ubu2 /tmp/tmp.WKMGJYxSdl $ ls -alh
total 48K
drwx------  3 zmully zmully 4.0K Sep  9 16:06 .
drwxrwxrwt 36 root   root    36K Sep  9 16:06 ..
drwxrwxr-x  2 zmully zmully 4.0K Sep  9 16:06 cdktf
zmully@ubu2 /tmp/tmp.WKMGJYxSdl $ npx projen new cdktf-construct --outdir cdktf
👾 Project definition file was created at /tmp/tmp.WKMGJYxSdl/cdktf/.projenrc.js
yarn install v1.22.19
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 23.14s.
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 5.79s.
Initialized empty Git repository in /tmp/tmp.WKMGJYxSdl/.git/
[main (root-commit) 7e812a2] chore: project created with projen
 21 files changed, 7967 insertions(+)
 create mode 100644 cdktf/.eslintrc.json
 create mode 100644 cdktf/.gitattributes
 create mode 100644 cdktf/.github/pull_request_template.md
 create mode 100644 cdktf/.github/workflows/build.yml
 create mode 100644 cdktf/.github/workflows/pull-request-lint.yml
 create mode 100644 cdktf/.github/workflows/release.yml
 create mode 100644 cdktf/.github/workflows/upgrade-main.yml
 create mode 100644 cdktf/.gitignore
 create mode 100644 cdktf/.mergify.yml
 create mode 100644 cdktf/.npmignore
 create mode 100644 cdktf/.projen/deps.json
 create mode 100644 cdktf/.projen/files.json
 create mode 100644 cdktf/.projen/tasks.json
 create mode 100644 cdktf/.projenrc.js
 create mode 100644 cdktf/LICENSE
 create mode 100644 cdktf/README.md
 create mode 100644 cdktf/package.json
 create mode 100644 cdktf/src/index.ts
 create mode 100644 cdktf/test/hello.test.ts
 create mode 100644 cdktf/tsconfig.dev.json
 create mode 100644 cdktf/yarn.lock
zmully@ubu2 /tmp/tmp.WKMGJYxSdl (main) $ ls -alh
total 52K
drwx------  4 zmully zmully 4.0K Sep  9 16:07 .
drwxrwxrwt 36 root   root    36K Sep  9 16:07 ..
drwxrwxr-x  7 zmully zmully 4.0K Sep  9 16:07 cdktf
drwxrwxr-x  8 zmully zmully 4.0K Sep  9 16:07 .git


```

### PR DESCRIPTION
**Pull request by jolo-dev**
This PR fixes in #2099 non-working `--outdir` - option.
The problem was that `writeFileSync` creates a file only but requires that the folder which is passed by `--outdir` already exists.

Here we switched the lib from `fs` to `fs-extra` and changed to `outputFileSync` instead of `writeFileSync`.

fixes #2099 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mrgrain** [comment]
Thanks for that fix @jolo-dev 😃
Would you mind adding a test case for this scenario do we don't lodging support for it in future?

**Comment by jolo-dev** [comment]
> Thanks for that fix @jolo-dev 😃 Would you mind adding a test case for this scenario do we don't lodging support for it in future?

Hey @mrgrain, Hope that is fine. Thanks for your reply :)

**Comment by mrgrain** [comment]
Thanks @jolo-dev that looks good. I'll have time later to give it a proper review later.

By the way, the test setup has some sort of cleanup management built-in. 

Have a look how this test does it: https://github.com/projen/projen/blob/main/test/typescript/typescript.test.ts#L17

**Comment by jolo-dev** [comment]
Hey @mrgrain,
Only for Typescript, right?
Because I cannot fully follow what I should do here?

File: test/java/projenrc.test.ts [hunk 1]
```diff
@@ -180,3 +182,26 @@ test("assert getJavaImport returns the correct import with no submodules", () =>
...
+    version: "1.2.3",
+  });
+  const projen = new Projenrc(project, pom);
+  const newOutDir = path.join(workDir, "foo");
+
+  expect(generateProjenrcSpy).toBeCalled();
+  expect(projen.project.outdir).toEqual(newOutDir);
+
+  removeSync(newOutDir); // otherwise we would accidentally check in the foo- directory
+});
```
Comment (mrgrain) [inline test/java/projenrc.test.ts:207]: @jolo-dev There are various helpers available, but I think this is the most readable one:
```suggestion
test("assert generateProjenrc returns the correct projenrc with correct outdir", () => {
  withProjectDir((projectdir) => {
    const newOutDir = path.join(projectdir, "foo");
    const project = new TestProject(
      renderProjenInitOptions("projen.java.JavaProject", {
        outdir: newOutDir,
      })
    );
    const pom = new Pom(project, {
      groupId: "my.group.id",
      artifactId: "hello-world",
      version: "1.2.3",
    });
    const projen = new Projenrc(project, pom);
    expect(projen.project.outdir).toEqual(newOutDir);
  });
});
```

**Review by mrgrain — APPROVED** [review]
Let's get this merged. I'm going to fix up the build failure,  that was on me.

**Comment by jolo-dev** [comment]
@mrgrain Ah nice. Thank you :)
Will you take it from here?

**Comment by mrgrain** [comment]
> @mrgrain Ah nice. Thank you :)
> Will you take it from here?

Yes I'll take care of it! Thanks again for your contribution and sorry for the delay.

**Pull request merged**